File: Data Structures & Algorithms/maximum-product-subarray/submission-1.go
func maxProduct(nums []int) int {
    curMax := nums[0]
    curMin := nums[0]
    ans := nums[0]
    for _,x := range nums[1:]{
        if x<0 {
            curMax,curMin=curMin,curMax
        }
        curMax=max(x,x*curMax)
        curMin=min(x,x*curMin)
        ans=max(ans,curMax)
    }
    return ans
}
func max(a, b int) int {
    if a > b {
        return a
    }
    return b
}

func min(a, b int) int {
    if a < b {
        return a
    }
    return b
}

File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-1.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        count = {}
        l=0
        cnt=0
        ans=0
        for r in range(len(s)):
            count[s[r]]=count.get(s[r],0)+1
            while(count[s[r]]>1):
                count[s[l]] -= 1
                l += 1                
            ans = max(ans, r - l + 1)
        return ans